### DISCUSSION TITLE
Vercel Express.js Deployment

### DISCUSSION THREAD

**Discussion by OrkhanGG** [vercel/turborepo#9657] [Help]
### Summary

Hello, everyone! 👋

I'm looking for some guidance on deploying an Express.js app as part of a monorepo on Vercel using Turborepo. Despite trying various approaches, I haven’t had much success.

Here’s what I’ve tried so far:

Configured custom build commands to focus on building only the backend project for deployment.
Attempted to deploy the backend individually within the monorepo structure.
Unfortunately, neither approach has worked. While I understand there’s a lot of information available online, I couldn’t find any dedicated resources or documentation specific to this use case.

I’m sure many others have faced this challenge, and I’d appreciate any general guidance or best practices you could share to help me and others in the community.

Thanks in advance! 🙏

**Comment by codebykarim** [comment]
# Simple Express.js Deployment on Vercel

This guide shows how to deploy an Express.js application on Vercel while using internal packages in a monorepo setup.

## Setup

Create these 4 files in your `apps/backend` folder:

### 1. `api/index.ts`
```typescript
import express from "express";
import { log } from "@repo/logger";
import { json, urlencoded } from "body-parser";

const app = express();

app.use(urlencoded({ extended: true }));
app.use(json());

app.get("/status", (req, res) => {
  log("status");
  res.json({ message: "Express on Vercel" });
});

app.listen(3000, () => log("Server ready on port 3000."));

module.exports = app;
```

### 2. `package.json`
```
{
  "name": "backend",
  "version": "1.0.0",
  "main": "index.ts",
  "scripts": {
    "start": "node src/index.ts"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": "",
  "dependencies": {
    "@repo/logger": "workspace:*",
    "body-parser": "^1.20.3",
    "express": "^5.0.1"
  },
  "devDependencies": {
    "@types/body-parser": "^1.19.5",
    "@types/express": "4.17.17",
    "@types/node": "^22.12.0",
    "tsup": "^8.3.6"
  }
}
``` 
### 2. `turbo.json`
```
{
  "extends": ["//"],
  "tasks": {
    "build": {
      "env": ["PORT"],
      "outputs": ["dist/**"]
    }
  }
}

``` 
### 2. `vercel.json`
```
{ "version": 2, "rewrites": [{ "source": "/(.*)", "destination": "/api" }] }


```
